UnitySetup/UnitySetup.psd1

```diff
@@ -1,158 +1,159 @@
+
+    # Supported PSEditions
+    # CompatiblePSEditions = @()
+
+    # ID used to uniquely identify this module
+    GUID              = '6dc524ef-5f56-4bc3-b04d-3c2906898f40'
+
+    # Author of this module
+    Author            = 'Josh Wittner'
+
+    # Company or vendor of this module
+    CompanyName       = 'Microsoft'
+
+    # Copyright statement for this module
+    Copyright         = 'Copyright (c) Microsoft Corporation. All rights reserved.'
+
+    # Description of the functionality provided by this module
+    Description       = 'Tools for managing and automating your Unity installs and projects.'
+
+    # Minimum version of the Windows PowerShell engine required by this module
+    # PowerShellVersion = ''
+
+    # Name of the Windows PowerShell host required by this module
+    # PowerShellHostName = ''
+
+    # Minimum version of the Windows PowerShell host required by this module
+    # PowerShellHostVersion = ''
+
+    # Minimum version of Microsoft .NET Framework required by this module. This prerequisite is valid for the PowerShell Desktop edition only.
+    # DotNetFrameworkVersion = ''
+
+    # Minimum version of the common language runtime (CLR) required by this module. This prerequisite is valid for the PowerShell Desktop edition only.
+    # CLRVersion = ''
+
+    # Processor architecture (None, X86, Amd64) required by this module
+    # ProcessorArchitecture = ''
+
+    # Modules that must be imported into the global environment prior to importing this module
+    RequiredModules   = @(
+        @{ModuleName = "powershell-yaml"; ModuleVersion = "0.3"; Guid = "6a75a662-7f53-425a-9777-ee61284407da" }
+    )
+
+    # Assemblies that must be loaded prior to importing this module
+    # RequiredAssemblies = @()
+
+    # Script files (.ps1) that are run in the caller's environment prior to importing this module.
+    ScriptsToProcess  = @()
+
+    # Type files (.ps1xml) to be loaded when importing this module
+    # TypesToProcess = @()
+
+    # Format files (.ps1xml) to be loaded when importing this module
+    # FormatsToProcess = @()
+
+    # Modules to import as nested modules of the module specified in RootModule/ModuleToProcess
+    # NestedModules = @()
+
+    # Functions to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no functions to export.
+
+    FunctionsToExport = @(
+        'Find-UnitySetupInstaller',
+        'Select-UnitySetupInstaller',
+        'Test-UnitySetupInstance',
+        'Get-UnityProjectInstance',
+        'Test-UnityProjectInstanceMetaFileIntegrity',
+        'Get-UnitySetupInstance',
+        'Request-UnitySetupInstaller',
+        'Install-UnitySetupInstance',
+        'Select-UnitySetupInstance',
+        'Uninstall-UnitySetupInstance',
+        'Get-UnityEditor',
+        'Start-UnityEditor',
+        'ConvertTo-UnitySetupComponent',
+        'Get-UnityLicense',
+        'Update-UPMConfig'
```

Comment:
By the way, out of the new CMDLets , do you think any of them are worth exporting aside from `Update-UPMConfig`?   I don't necessarily think they have utility independently, but should they be exported as a matter of policy?

Reply:
Yes, this PS module is used by a lot of people so we should think about our solutions as being used by people outside of our use cases. After we're done anyone should be able to load a UPMConfig and a ProjectManifest and do whatever they want with them and then save them back out. We'll focus on making it easy to do to them what we need, but the general import, modify, export should be supported.

So we should export all the cmdlets for those operations around UPMConfig and ProjectManifest.

Reply:
Well, the manifest is JSON so I think `ConvertFrom-Json` for import and then `ConvertTo-Json` for export makes a lot of sense there. For the upm config I think we continue to make custom objects for now (since we know the format of Unity's toml here and don't need something more general) - there is at least one decent looking [OSS toml parser](https://github.com/jborean93/PSToml) for PS, but I don't think we need to bite that off yet, we can stay closer to what we have support for in the code already.